### PR TITLE
add dubbo wire protocol for pu server

### PR DESCRIPTION
**Pull request by TrueAbc**
## What is the purpose of the change
1. add dubbo detector implemention
2.  make dubbo protocol work well using pu server

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyPortUnificationServerHandler.java [hunk 1]
```diff
@@ -49,16 +53,18 @@ public class NettyPortUnificationServerHandler extends ByteToMessageDecoder {
     private final ChannelHandler handler;
     private final boolean detectSsl;
     private final List<WireProtocol> protocols;
+    private Map<String, Channel> dubboChannels;
```
Comment (CrazyHZM) [inline dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyPortUnificationServerHandler.java:56]: The duties of `dubboChannels` and `ChannelGroup` are the same. Need to consider merging.

File: dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DubboWireProtocol.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+public class DubboWireProtocol extends AbstractWireProtocol {
+    public DubboWireProtocol() {
+        super(new DubboDetector());
+    }
+
+
+    @Override
+    public void configServerProtocolHandler(URL url, ChannelOperator operator) {
+        List<ChannelHandler> handlers = new ArrayList<>();
+        operator.configChannelHandler(handlers);
```
Comment (CrazyHZM) [inline dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DubboWireProtocol.java:38]: Add comment why is the handler empty

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

File: dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyPortUnificationServerHandler.java (hunk 1 above)
Comment (TrueAbc) [reply dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyPortUnificationServerHandler.java:56]: dubboChannels can make pu server manage channels defined in dubbo, and this ablity is needed by some test cases. So I delete DefaultChannelGroup and choose dubboChannels

**Comment by TrueAbc** [comment]
changes of this pr is moved to https://github.com/apache/dubbo/pull/10602

**Pull request closed without merging**